packages/react/src/components/Listbox/Listbox.tsx

```diff
@@ -16,34 +16,43 @@ import useSharedRef from '../../utils/useSharedRef';
 
 const keys = ['ArrowUp', 'ArrowDown', 'Home', 'End', 'Enter', ' '];
 
-interface ListboxProps
-  extends PolymorphicProps<
-    Omit<React.HTMLAttributes<HTMLElement>, 'onSelect'>
-  > {
-  value?: ListboxValue;
+type ListboxProps = PolymorphicProps<
+  Omit<React.HTMLAttributes<HTMLElement>, 'defaultValue' | 'onSelect'>
+> & {
   navigation?: 'cycle' | 'bound';
-  onSelectionChange?: <T extends HTMLElement = HTMLElement>({
-    value
-  }: {
-    target: T;
-    previousValue: ListboxValue;
-    value: ListboxValue;
-  }) => void;
   onActiveChange?: (option: ListboxOption) => void;
-}
+  onSelectionChange?: <T extends HTMLElement = HTMLElement>(selection: {
+    target: T | undefined;
+    previousValue: ListboxValue | ListboxValue[];
+    value: ListboxValue | ListboxValue[];
+  }) => void;
+} & (
+    | {
```

Comment:
I'm not sure the discriminated union buys a whole lot here, although if we did want a discriminated union I would make this more readable by keeping the existing type and unioning the types below where needed (example):

```tsx
interfact ListboxProps  = ....
type MultiselectListBoxProps = Omit<ListboxProps, 'value', 'defaultValue', 'multiselect'> & {
  value?: ListboxValue | ListboxValue[]
  defaultValue?: ListboxValue | ListboxValue[]
  multiselect: true
}
```

Part of the reasoning behind allowing array and non-array values is that it allows the component to be more flexible when specifying the value. When referenced, we would use `ListboxProps | MultiselectListboxProps`.

Reply:
By doing this I'm only allowing `value: string[]` when `multiselect = true` and `value: string` when `multiselect = false`. I don't really understand why we should allow value to be both string and string[] when multiselect is true, behavior will just be broken in my opinion. But if that's required, I'll change that.

Reply:
Part of the reasoning would be to allow for more graceful inputs since we can't necessarily enforce that everyone is using typescript. Someone using `<Combobox value="foo" multiselect />` at a minimum should not cause an error although we could potentially log a warning if this wasn't the desired interface.